frontend/lib/screens/transaction_details_screen.dart

```diff
@@ -46,33 +50,80 @@ class TransactionDetailsScreenState extends State<TransactionDetailsScreen> {
-              backgroundColor: Utils.mediumLightColor,
-              foregroundColor: Colors.black,
-              leading: IconButton(
-                  onPressed: (() {
-                    Navigator.pop(context);
-                  }),
-                  icon: Icon(Icons.arrow_back)),
-              title: headerInfo(),
-              centerTitle: false,
-              elevation: 0,
-            ),
-            body:
-                ListView(children: [buildDataTable(), deleteButton(context)])));
+            body: NestedScrollView(
+                headerSliverBuilder: ((context, innerBoxIsScrolled) => [
+                      SliverAppBar(
+                        expandedHeight: 250,
+                        backgroundColor: Utils.mediumLightColor,
+                        foregroundColor: Colors.black,
+                        leading: Container(
+                            padding: EdgeInsets.only(top: 10, left: 10),
+                            child: IconButton(
+                                alignment: Alignment.topCenter,
+                                onPressed: (() {
+                                  Navigator.of(context)
+                                      .popUntil((route) => route.isFirst);
+                                }),
+                                icon: Icon(Icons.arrow_back))),
+                        actions: [
+                          Container(
+                            padding: EdgeInsets.only(top: 10, right: 5),
+                            child: ElevatedButton(
+                                style: ElevatedButton.styleFrom(
+                                  shape: CircleBorder(),
+                                  alignment: Alignment.center,
+                                  backgroundColor:
+                                      Utils.lightColor.withOpacity(0.9),
+                                  foregroundColor: Utils.darkColor,
+                                ),
+                                onPressed: (() {
+                                  widget.transaction.receiptID != null
+                                      ? Text("TODO: Show image")
+                                      : receiptBtnAlertDialog(context);
+                                }),
+                                child: Icon(Icons.receipt_long_rounded)),
+                          )
+                        ],
+                        flexibleSpace: FlexibleSpaceBar(
+                          background: Container(
+                              decoration: BoxDecoration(
+                                  gradient: LinearGradient(
+                                      begin: Alignment.bottomLeft,
+                                      end: Alignment.topRight,
+                                      colors: [
+                                Utils.mediumLightColor,
+                                Utils.mediumDarkColor
+                              ]))),
+                          title: headerInfo(),
```

Comment:
Having all the info in the title makes the scroll-induced text shrinkage look rather odd. I think it would be neater if scrolling caused the receipt info to overlap the header info until the actual title is reached. It would also be nice if the receipt button could be aligned with some element of the header, since it currently looks like it floats above the header.